JavaScript/JavaScriptSDK/SendBuffer.ts

```diff
@@ -183,6 +187,16 @@ module Microsoft.ApplicationInsights {
             var sentElements = this.getBuffer(SessionStorageSendBuffer.SENT_BUFFER_KEY);
             sentElements = sentElements.concat(payload);
 
+            if (sentElements.length > SessionStorageSendBuffer.MAX_BUFFER_SIZE) {
```

Comment:
Per discussion we'll leave it like this for code efficiency and trust that the limits really shouldn't be hit in normal circumstances.